kuma/core/middleware.py

```diff
@@ -8,82 +7,203 @@
 from django.http import (HttpResponseForbidden,
                          HttpResponsePermanentRedirect,
                          HttpResponseRedirect)
+from django.middleware.locale import (
+    LocaleMiddleware as DjangoLocaleMiddleware)
 from django.utils import translation
 from django.utils.encoding import iri_to_uri, smart_str
+from django.utils.six.moves.urllib.parse import (
+    urlencode, urlsplit, urlunsplit)
 from whitenoise.middleware import WhiteNoiseMiddleware
 
 from .decorators import add_shared_cache_control
-from .urlresolvers import Prefixer, set_url_prefixer, split_path
-from .utils import is_untrusted, urlparams
+from .i18n import django_language_code_to_kuma
+from .utils import is_untrusted
 from .views import handler403
 
 
-class LocaleURLMiddleware(object):
+class LangSelectorMiddleware(object):
     """
-    Based on zamboni.amo.middleware.
-    Tried to use localeurl but it choked on 'en-US' with capital letters.
+    Redirect requests with a ?lang= parameter.
 
-    1. Search for the locale.
-    2. Save it in the request.
-    3. Strip them from the URL.
+    This should appear higher than LocaleMiddleware in the middleware list.
     """
 
     def process_request(self, request):
-        prefixer = Prefixer(request)
-        set_url_prefixer(prefixer)
-        full_path = prefixer.fix(prefixer.shortened_path)
-        lang = request.GET.get('lang')
-
-        if lang in dict(settings.LANGUAGES):
-            # Blank out the locale so that we can set a new one. Remove lang
-            # from the query params so we don't have an infinite loop.
-            prefixer.locale = ''
-            new_path = prefixer.fix(prefixer.shortened_path)
-            query = dict((smart_str(k), v) for
+        """Redirect if ?lang query parameter is valid."""
+        query_lang = request.GET.get('lang')
+        if query_lang not in dict(settings.LANGUAGES):
+            return
```

Comment:
Would be helpful if possible to explicitly mention the type of return.
Maybe `None`?

Reply:
``return`` and ``return None`` have [the same effect](https://stackoverflow.com/a/15300671/10612), so I usually use the shorter one. If it makes the code clearer, I can use ``return None`` or add a comment.